src/dataflow/src/operator.rs

```diff
@@ -261,32 +272,6 @@ where
         })
     }
 
-    fn filter_fallible<E, L>(&self, mut logic: L) -> (Stream<G, D1>, Stream<G, E>)
-    where
-        E: Data,
-        L: FnMut(&D1) -> Result<bool, E> + 'static,
-    {
-        let mut storage = Vec::new();
-        self.unary_fallible(Pipeline, "FilterFallible", move |_, _| {
-            Box::new(move |input, ok_output, err_output| {
-                input.for_each(|time, data| {
-                    let mut ok_session = ok_output.session(&time);
-                    let mut err_session = err_output.session(&time);
-                    data.swap(&mut storage);
-                    storage.retain(|d1| match logic(d1) {
-                        Ok(retain) => retain,
-                        Err(e) => {
-                            err_session.give(e);
-                            false
-                        }
-                    });
-                    if !storage.is_empty() {
-                        ok_session.give_vec(&mut storage);
```

Comment:
The new implementation of this method ends up using `ok_session.give(datum)` instead of `ok_session.give_vec(&mut storage)`. Could that have any performance implications? Does that add any more memory copies that were avoided at the cost of this code redundancy?

Reply:
I think it has minor implications that I am happy to absorb responsibility for. :D  It means that there may be one more `memcpy` per batch of data that arrive, vs a pointer swap. But tbh I'm ok with it in the interest of reducing the code down, until the point that we can measure something like a memcpy.

Reply:
Actually, I take it bake there may be no regression, because we now avoid the copy into `storage` that we were doing to be able to pluck out errors. Seems all good to me.